lib/requestwrapper.ts

```diff
@@ -20,23 +20,276 @@ import FormData = require('form-data');
 import https = require('https');
 import querystring = require('querystring');
 import { PassThrough as readableStream } from 'stream';
+import {HeaderOptions} from "./base_service";
 import { buildRequestFileObject, getMissingParams, isEmptyObject, isFileParam } from './helper';
 
 // tslint:disable-next-line:no-var-requires
 const pkg = require('../package.json');
 const isBrowser = typeof window === 'object';
 const globalTransactionId = 'x-global-transaction-id';
 
-// axios sets the default Content-Type for `post`, `put`, and `patch` operations
-// to 'application/x-www-form-urlencoded'. This causes problems, so overriding the
-// defaults here
-['post', 'put', 'patch'].forEach(method => {
-  axios.defaults.headers[method]['Content-Type'] = 'application/json';
-});
+const allowedAxiosConfig = ['transformRequest', 'transformResponse', 'paramsSerializer', 'paramsSerializer', 'timeout', 'withCredentials', 'adapter', 'responseType', 'responseEncoding', 'xsrfCookieName', 'xsrfHeaderName', 'onUploadProgress', 'onDownloadProgress', 'maxContentLength', 'validateStatus', 'maxRedirects', 'socketPath', 'httpAgent', 'httpsAgent', 'proxy', 'cancelToken'];
```

Comment:
So I don't _love_ the idea of maintaining a list of axios request params in this class. If we don't want certain things to make it in the request, I would rather blacklist them than whitelist all of these.

That said, if you think this is best I'm fine to leave as is. Maybe throw a comment in there saying why you're doing it that way